# Práctica 1.1 — Probabilidad y Estadística- Análisis de la evolución poblacional mundial
# Juan Carlos Orozco Nieto

Actividad 1 - Markdown (Ver el origen de los datos)

Fuente:
> UN, World Population Prospects (2024) – processed by Our World in Data. “Annual change in population – UN WPP” [dataset]. United Nations, “World Population Prospects”; United Nations, “World Population Prospects - Interim Update” [original data].

Preguntas
1. ¿Quién produjo originalmente los datos? La Organización de las Naciones Unidas (ONU), en su reporte World Population Prospects (2024).
2. ¿Qué organización los procesa para su utilización en Our World in Data? Our World in Data (OWID) toma los datos originales de la ONU y los procesa para publicarlos.
3. ¿Qué representa la variable que analizarás? La variable "Annual change in population" representa el cambio anual de población, incluyendo efectos como nacimientos, muertes y migración.

In [1]:
# Actividad 2 - Código

import pandas as pd
import numpy as np

df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

print("-- Primeras filas:\n", df.head())
print("\n-- Dimensiones (filas, columnas):", df.shape)
print("\n-- Nombres de columnas:", df.columns.tolist())
print("\n-- Tipos de datos:\n", df.dtypes)
print("\n-- Descripciones:", df.describe())
print("\n-- Valores faltantes por columna:", df.isnull().sum())
print("\n-- Cantidad de entidades distintas:", df["Entity"].nunique())

# Descripción de lo realizado: Importar csv y usar funciones como head, shape, tolist, etc. para ver datos generales sobre el acomodo, tipos y cantidad de los datos del csv

-- Primeras filas:
         Entity Code  Year  Annual change in population  \
0  Afghanistan  AFG  1951                     103163.0   
1  Afghanistan  AFG  1952                     108441.0   
2  Afghanistan  AFG  1953                     108919.0   
3  Afghanistan  AFG  1954                     111251.0   
4  Afghanistan  AFG  1955                     119027.0   

   Annual population change (Projected)  
0                                   NaN  
1                                   NaN  
2                                   NaN  
3                                   NaN  
4                                   NaN  

-- Dimensiones (filas, columnas): (38400, 5)

-- Nombres de columnas: ['Entity', 'Code', 'Year', 'Annual change in population', 'Annual population change (Projected)']

-- Tipos de datos:
 Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float

Actividad 2 - Markdown
1.	¿Cuántas filas y columnas tiene el dataset? 38400 filas y 5 columnas
2.	¿Qué tipo de datos contiene la variable Year? int64
3.	¿Existen valores faltantes? Si (Code: 1200, Annual change in population: 19712, Annual population change: 18688)
4.	¿Cuántas entidades diferentes contiene el dataset? 256


In [2]:
# Actividad 3 - Código
anio = 2015
df_anio = df[df["Year"] == anio] #filtrar por año
cambio_poblacion = df_anio["Annual change in population"] #asignar los datos de la columna a una variable

print("-- Media:", cambio_poblacion.mean())
print("-- Mediana:", cambio_poblacion.median())
print("-- Varianza:", cambio_poblacion.var())
print("-- Desviación estándar:", cambio_poblacion.std())
print("-- Mínimo:", cambio_poblacion.min())
print("-- Máximo:", cambio_poblacion.max())
print("\n-- Entidad con el valor mínimo:\n", df_anio[cambio_poblacion == cambio_poblacion.min()])
print("\n-- Entidad con el valor máximo:\n", df_anio[cambio_poblacion == cambio_poblacion.max()])

# Descripción de lo realizado: usar funciones como mean, median, etc. para calcular datos en base a una una columna obtenida de un año específico

-- Media: 2437062.7890625
-- Mediana: 56624.0
-- Varianza: 116931073859650.22
-- Desviación estándar: 10813467.245044496
-- Mínimo: -849046.0
-- Máximo: 88875637.0

-- Entidad con el valor mínimo:
       Entity Code  Year  Annual change in population  \
33514  Syria  SYR  2015                    -849046.0   

       Annual population change (Projected)  
33514                                   NaN  

-- Entidad con el valor máximo:
       Entity      Code  Year  Annual change in population  \
37864  World  OWID_WRL  2015                   88875637.0   

       Annual population change (Projected)  
37864                                   NaN  


Actividad 3 - Markdown
1. ¿Cuál es el valor de la media? 2437062.79
2. ¿Cuál es el valor de la mediana? 56624
3. ¿Cuál es el valor de la desviación estándar? 10813467.24
4. ¿Qué entidad presenta el valor mínimo y cuál el máximo? Syria (-849046) y World (88875637)

In [3]:
# Actividad 4 - Código

import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

# Descripción de lo realizado: únicamente verificar versiones de lo usado en esta práctica

Python: 3.13.2 (tags/v3.13.2:4f8bb39, Feb  4 2025, 15:23:48) [MSC v.1942 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


In [4]:
# Filtros de clase

print(df.iloc[0:5, 0:4])
print(df.iloc[:, [0,2,3]])
print(df[df['Entity'] == 'Mexico'])
print(df[df["Year"]==2020])
df_2006_2020_mex = df[(df["Entity"] == "Mexico") & (df["Year"]>=2006) & (df["Year"]<=2020)]
print(df_2006_2020_mex[["Entity", "Year", "Annual change in population"]])
print(df[["Entity"]])
print(df.iloc[30:35])
df.loc[:,["Entity","Year","Annual change in population"]]
df.loc[(df["Entity"]=="Mexico") & (df["Year"]==2020)]
df.loc[0:5]
df.iloc[:, 0:3]

        Entity Code  Year  Annual change in population
0  Afghanistan  AFG  1951                     103163.0
1  Afghanistan  AFG  1952                     108441.0
2  Afghanistan  AFG  1953                     108919.0
3  Afghanistan  AFG  1954                     111251.0
4  Afghanistan  AFG  1955                     119027.0
            Entity  Year  Annual change in population
0      Afghanistan  1951                     103163.0
1      Afghanistan  1952                     108441.0
2      Afghanistan  1953                     108919.0
3      Afghanistan  1954                     111251.0
4      Afghanistan  1955                     119027.0
...            ...   ...                          ...
38395     Zimbabwe  2096                          NaN
38396     Zimbabwe  2097                          NaN
38397     Zimbabwe  2098                          NaN
38398     Zimbabwe  2099                          NaN
38399     Zimbabwe  2100                          NaN

[38400 rows x 3 colum

,Entity,Code,Year
0,Afghanistan,AFG,1951
1,Afghanistan,AFG,1952
2,Afghanistan,AFG,1953
3,Afghanistan,AFG,1954
4,Afghanistan,AFG,1955
...,...,...,...
38395,Zimbabwe,ZWE,2096
38396,Zimbabwe,ZWE,2097
38397,Zimbabwe,ZWE,2098
38398,Zimbabwe,ZWE,2099
